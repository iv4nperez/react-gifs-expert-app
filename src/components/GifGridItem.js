import React from 'react'

export const GifGridItem = ({ id, title, url }) => {

    return (
        <div className="card animate__animated animate__fadeIn">
           <img className="cover" src={ url } alt={ title } />
           <p>{ title }</p>
        </div>
    )
}
