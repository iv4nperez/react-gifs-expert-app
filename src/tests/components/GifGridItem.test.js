import React from 'react';
import  "@testing-library/jest-dom";
import { shallow } from "enzyme";
import { GifGridItem } from '../../components/GifGridItem';

describe('Pruebas al componente GifGridItem', () => {
    const title ='Un titulo';
    const url   ='https//localhost:20000'
    let wrapper = shallow( <GifGridItem title={ title } url={ url }  /> );
    
    test('debe de mostrar correctamente el componente <GifGridItem/> ', () => {
        expect( wrapper ).toMatchSnapshot();
    });

    test('debe de tener un pÃ¡rrafo con el titulo', () => {
       const p = wrapper.find('p');
       expect( p.text().trim() ).toBe( title );
    });
    
    test('debe de ser una imagen, tener un src y un alt asignado', () => {
        const img = wrapper.find('img');
        expect( img.prop('src')).toBe( url );
        expect( img.prop('alt')).toBe( title );
    });
    
    test('debe de tener la clase animate__fadeIn', () => {
    
        const div = wrapper.find('div');
        const className = div.prop('className');
        expect( className.includes( 'animate__fadeIn' ) ).toBe( true );
    });
    
    
    
});
