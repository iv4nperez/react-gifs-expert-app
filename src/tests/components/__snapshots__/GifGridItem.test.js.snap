// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas al componente GifGridItem debe de mostrar correctamente el componente <GifGridItem/>  1`] = `
<div
  className="card animate__animated animate__fadeIn"
>
  <img
    alt="Un titulo"
    className="cover"
    src="https//localhost:20000"
  />
  <p>
    Un titulo
  </p>
</div>
`;
